devicetypes/smartthings/ezex-smart-electric-switch.src/ezex-smart-electric-switch.groovy

```diff
@@ -0,0 +1,123 @@
+    }
+
+    tiles(scale: 2){
+        multiAttributeTile(name:"switch", type: "generic", width: 6, height: 4, canChangeIcon: true){
+                tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+                        attributeState("on", label: '${name}', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#00a0dc")
+                        attributeState("off", label: '${name}', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff")
+                }
+        }
+        valueTile("power", "device.power", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} W'
+        }
+        valueTile("energy", "device.energy", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} kWh'
+        }
+        standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'reset kWh', action:"reset"
+        }
+
+        main(["switch"])
+        details(["switch","power","energy","refresh","reset"])
+    }
+}
+
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        log.info event
+        sendEvent(event)
+    } else {
+        List result = []
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "Desc Map: $descMap"
+        
+        List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
+        descMap.additionalAttrs.each {
+            attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+        }
+        attrData.each {
+                def map = [:]
+                if (it.clusterInt == 0x0702 && it.attrInt == 0x0400) {
+                        log.debug "power"
+                        map.name = "power"
+                        map.value = zigbee.convertHexToInt(it.value)/1000
+                }
+                if (it.clusterInt == 0x0702 && it.attrInt == 0x0000) {
+                         log.debug "energy"
+                         map.name = "energy"
+                         map.value = zigbee.convertHexToInt(it.value)/1000000
+                }
+
+                if (map) {
+                        result << createEvent(map)
+                }
+                log.debug "Parse returned $map"
+        }
+        return result       
+    }
+}
+
+def off() {
+        zigbee.off()
+}
+
+def on() {
+        zigbee.on()
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    return refresh()
+}
+
+def installed() {
+    return refresh()
```

Comment:
this needs to be `response(refresh())` because of how the messages returned from `installed()` are treated.

Reply:
Or just removed, since it is called correctly in `configure()`